reference/src/layout/structs-and-tuples.md

```diff
@@ -141,6 +141,22 @@ struct Zst2(Zst1, Zst0);
 # }
 ```
 
+#### Structs with 1-ZST fields
+
+For the purposes of struct layout [1-ZST] fields are ignored.
```

Comment:
I just noticed there's a "Zero-sized structs" section just above. This has strong overlap now.

We *could* collapse that other section to just be about "structs with no fields". That nicely combines with this section for structs consisting only of 1-ZST. However, that would reduce what we say (e.g. about structs consisting only of ZST but some having non-trivial alignment requirements). Do we care? It would also simplify the documentation which is nice.

Reply:
I realized the overlap as well, but thought about:

```rust
type Zst2 = [u16; 0];
struct S(Zst2);
```

which has zero-size but is not covered by the "we ignore 1-ZST"s clause. 

I decided to keep them as separate subsections on purpose, at least until someone figures out a good way to combine them.

Reply:
I'd personally prefer to cut the other section until we find a nice way to expand what we got. ;)

But I can see how others might have a different opinion here.

But at least, I think it is worth adding a note like "in particular, a struct with *no* fields is a ZST, and if it has no `repr` attribute it is moreover a 1-ZST as it also has no alignment requirements".

Reply:
I'd rather not loose anything that we have already guaranteed. Each guarantee takes too long to just throw them away for an editorial change. 

I'm all in from restructuring the text as much as time allows once its in, but those should be non-functional changes.

Reply:
Fair.